backend/de.metas.contracts/src/main/java/de/metas/contracts/invoicecandidate/FlatrateTerm_Handler.java

```diff
@@ -194,12 +198,19 @@ public void setOrderedData(@NonNull final I_C_Invoice_Candidate ic)
 		ic.setQtyEntered(calculateQtyOrdered.toBigDecimal());
 		ic.setC_UOM_ID(calculateQtyOrdered.getUomId().getRepoId());
 
-		final ProductId productId = ProductId.ofRepoId(term.getM_Product_ID());
+		if (term.getType_Conditions().equals(X_C_Flatrate_Term.TYPE_CONDITIONS_ModularContract))
+		{
+			ic.setQtyOrdered(calculateQtyOrdered.toBigDecimal());
+		}
+		else
+		{
+			final ProductId productId = ProductId.ofRepoId(term.getM_Product_ID());
 
-		final IUOMConversionBL uomConversionBL = Services.get(IUOMConversionBL.class);
+			final IUOMConversionBL uomConversionBL = Services.get(IUOMConversionBL.class);
```

Comment:
make uomConversionBL a class field

Reply:
also, why we avoid conversion in case it's a modular contract?
AFAIK, qtyOrdered shall ALWAYS be in product's UOM.

Reply:
clarified, will keep the conversion in case of a modular contract as well